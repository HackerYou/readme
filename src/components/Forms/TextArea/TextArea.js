import React from 'react';
import PropTypes from 'prop-types';

const TextArea = ({ label, name, cols, rows, handleChange }) => (
    <span>
        <label htmlFor={name}>{label}</label>
        <textarea cols={cols} rows={rows} name={name} id={name} onChange={handleChange} />
    </span>
);

export default TextArea;

TextArea.propTypes = {
    handleChange: PropTypes.func.isRequired,
    cols: PropTypes.string.isRequired,
    rows: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
};
