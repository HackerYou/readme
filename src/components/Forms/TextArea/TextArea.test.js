import React from 'react';
import TextArea from './TextArea';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('It renders the textarea component', () => {
    const props = {
        name: 'description',
        label: 'Dummy Textarea Component',
        cols: '30',
        rows: '10',
        handleChange: () => null,
        val: 'This is a dummy textarea component'
    };
    const component = renderer.create(<TextArea {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
