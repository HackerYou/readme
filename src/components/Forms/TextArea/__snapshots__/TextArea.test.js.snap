// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders the textarea component 1`] = `
<span>
  <label
    htmlFor="description"
  >
    Dummy Textarea Component
  </label>
  <textarea
    cols="30"
    id="description"
    name="description"
    onChange={[Function]}
    rows="10"
  />
</span>
`;
